src/cuda-poh-verify/poh_verify.cu

```diff
@@ -72,13 +74,90 @@ bool poh_init() {
     return success;
 }
 
+void save_input(uint8_t* hashes,
+    const uint64_t* num_hashes_arr,
+    size_t num_elems) {
+
+    FILE * fp;
+
+    const char *file_name = "test_hashes";
+    char temp_string[30];
+    sprintf(temp_string, "%s_%zu_%zu", file_name, index_file, num_elems);
+
+    fp = fopen (temp_string, "w");
+    if (fp == NULL) {
+        fprintf(stderr, "Could not create file %s\n",temp_string);
+        exit(-1);
+    }
+
+    FILE * fp2;
+
+    const char *file_name2 = "test_num_hashes_arr";
+    char temp_string2[25];
+    sprintf(temp_string2, "%s_%zu", file_name2, index_file);
+    fp2 = fopen (temp_string2, "w");
+    if (fp2 == NULL) {
+        fprintf(stderr, "Could not create file %s\n", temp_string2);
+        exit(-2);
+    }
+
+    FILE * fp3;
+
+    const char *file_name3 = "test_num_elems";
+    char temp_string3[25];
+    sprintf(temp_string3, "%s_%zu", file_name3, index_file);
+    fp3 = fopen (temp_string3, "w");
+    if (fp3 == NULL) {
+        fprintf(stderr, "Could not create file %s\n", file_name3);
+        exit(-3);
+    }
+
+    fprintf(fp3, "%lu", num_elems*SHA256_BLOCK_SIZE);
+    fclose(fp3);
+
+    for (size_t i = 0; i < num_elems*SHA256_BLOCK_SIZE; ++i) {
+        fprintf(fp, "%hhu ", hashes[i]);
+    }
+    fclose(fp);
+
+
+    for (size_t i = 0; i < num_elems; ++i) {
+        fprintf(fp2, "%lu ", num_hashes_arr[i]);
+    }
+    fclose(fp2);
+    index_file++;
+
+}
+
+void static inline save_out(uint8_t* hashes,
+    size_t num_elems) {
+
+    FILE * fp;
+
+    const char *file_name = "test_hashes_output";
+    char temp_string[25];
+    sprintf(temp_string, "%s_%zu", file_name, index_file-1);
+
+    fp = fopen (temp_string, "w");
+    if (fp == NULL) {
+        fprintf(stderr, "Could not create file %s\n", temp_string);
+        exit(-1);
+    }
+
+    for (size_t i = 0; i < num_elems*SHA256_BLOCK_SIZE; ++i) {
+        fprintf(fp, "%hhu ", hashes[i]);
+    }
+    fclose(fp);
+}
+
 extern "C" {
 int poh_verify_many(uint8_t* hashes,
                     const uint64_t* num_hashes_arr,
                     size_t num_elems,
                     uint8_t use_non_default_stream)
 {
     LOG("Starting poh_verify_many: num_elems: %zu\n", num_elems);
+    //save_input(hashes, num_hashes_arr, num_elems);
```

Comment:
can you revert this file? doesn't seem like any of the code added is actually used.

Reply:
Done.